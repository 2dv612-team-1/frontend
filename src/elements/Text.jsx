import styled from "styled-components";

const Text = styled.p`
  font-size: 16px;
  color: black;
  text-align: ${props => (props.center ? "center" : "left")};
`;

export default Text;
