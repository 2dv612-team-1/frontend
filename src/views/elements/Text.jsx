import styled from "styled-components";
import fonts from "../constants/fonts";

const Text = styled.p`
  font-size: 16px;
  color: ${props => (props.error ? "red" : props.success ? "green" : "black")};
  font-family: ${fonts.primary};
  margin-top: 0px;
`;

export default Text;
