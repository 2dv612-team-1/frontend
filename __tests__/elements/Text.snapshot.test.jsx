import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import "jest-styled-components";
import Text from "../../src/elements/Text";

describe("Text", () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Text {...props}>{props.children}</Text>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined,
      center: undefined
    };
    wrapper = undefined;
  });

  describe("when props are undefined", () => {
    it("Should match snapshot", () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it("should not be center", () => {
      const tree = Wrapper();
      expect(tree).toHaveStyleRule("text-align", "left");
    });
  });

  describe("when props are defined", () => {
    it("Should match snapshot", () => {
      props.children = "Text";
      props.center = true;
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it("should be center", () => {
      props.children = "Text";
      props.center = true;
      expect(Wrapper()).toHaveStyleRule("text-align", "center");
    });
  });
});
