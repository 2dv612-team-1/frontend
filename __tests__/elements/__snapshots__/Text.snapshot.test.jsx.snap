// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text when props are defined Should match snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: black;
  text-align: center;
}

<p
  className="c0"
>
  Text
</p>
`;

exports[`Text when props are undefined Should match snapshot 1`] = `
.c0 {
  font-size: 16px;
  color: black;
  text-align: left;
}

<p
  className="c0"
/>
`;
